cumulusci/tasks/salesforce/content_documents.py

```diff
@@ -0,0 +1,237 @@
+            "required": False,
+        },
+        "namespace_inject": {
+            "description": "If set, the namespace tokens in files and filenames are replaced with the namespace's prefix",
+            "required": False,
+        },
+    }
+
+    def _init_options(self, kwargs):
+        super()._init_options(kwargs)
+
+        path = Path(self.options.get("path"))
+
+        if not path.exists() or not path.is_file():
+            raise TaskOptionsError(
+                f'Invalid "path". No file found at {self.options["path"]}'
+            )
+
+        # Process queries into a list + inject namespaces into queries.
+        namespace = (
+            self.options.get("namespace_inject")
+            or self.project_config.project__package__namespace
+        )
+        if "managed" in self.options:
+            managed = process_bool_arg(self.options["managed"])
+        else:
+            managed = (
+                bool(namespace) and namespace in self.org_config.installed_packages
+            )
+        if "namespaced_org" in self.options:
+            namespaced_org = process_bool_arg(self.options["namespaced_org"])
+        else:
+            namespaced_org = bool(namespace) and namespace == self.org_config.namespace
+
+        queries = process_list_arg(self.options.get("queries") or [])
+        for i, query in enumerate(queries):
+            _, namespaced_query = inject_namespace(
+                "",
+                query,
+                namespace=namespace,
+                managed=managed,
+                namespaced_org=namespaced_org,
+            )
+            queries[i] = namespaced_query
+        self.options["queries"] = queries
+
+        # Set defaults.
+        self.options["share_type"] = self.options.get("share_type") or "I"
+
+        self.options["visibility"] = self.options.get("visibility") or "AllUsers"
+
+    def _insert_content_document(self) -> str:
+        path = Path(self.options["path"])
+
+        self.logger.info(f"Inserting ContentVersion from {path}")
+
+        # Any failures raise a SalesforceMalformedRequest instead of returning success as False.
+        result = self.sf.ContentVersion.create(
+            {
+                "PathOnClient": path.name,
+                "Title": path.stem,
+                "VersionData": base64.b64encode(path.read_bytes()).decode("utf-8"),
+            }
+        )
+        content_version_id = result["id"]
+
+        # Query the ContentDocumentId for our created record.
+        content_document_id = self.sf.query(
+            f"SELECT Id, ContentDocumentId FROM ContentVersion WHERE Id = '{content_version_id}'"
+        )["records"][0]["ContentDocumentId"]
+
+        self.logger.info(f'Success!  Inserted ContentDocument "{content_document_id}".')
+
+        return content_document_id
+
+    def _get_record_ids_to_link(self) -> list[str]:
```

Comment:
This type hinting syntax (using the collection directly instead of `typing.List`) is Python 3.9-only.

Reply:
Will update.